lib/command/Command.js

```diff
@@ -279,17 +279,29 @@ class Command {
         return true;
     }
 
-    process(args, msg) {
+    async process(args, msg) {
         var shouldDelete = this.deleteCommand && msg.channel.guild && msg.channel.permissionsOf(msg._client.user.id).has("manageMessages");
 
         if(this.hooks.preCommand) {
-            this.hooks.preCommand(msg, args);
+            if (this.hooks.preCommand(msg, args) !== undefined) {
```

Comment:
Since this is expected to return an object and objects are always truthy, couldn't you omit the `!== undefined` here?

(all these comments apply to all the hooks, just gonna make them once on this one)